ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/clientlibs/site/js/textinputview.js

```diff
@@ -77,6 +81,19 @@
             this.widget.addEventListener('focus', (e) => {
                 this.setActive();
             });
+            this.setupCharacterCounter();
+        }
+
+        setupCharacterCounter() {
+        	const counterSpan = this.getCounterSpan();
+        	const updateCharacterCounter = () => {
+        		const charCount = this.widget.value.length;
+        		if (counterSpan) {
+        			counterSpan.textContent = Granite.I18n.getMessage(charCount);
+        		}
```

Comment:
Also do the same for max character count, as authour can write a rule to modify that and then the modified value should show up and also will need to be localized.